nixos-tests/api.nix

```diff
@@ -0,0 +1,57 @@
+{
+  modules,
+  pkgs,
+}: let
+  inherit
+    (pkgs.lib)
+    mkDefault
+    ;
+
+  inherit
+    (builtins)
+    toString
+    ;
+
+  port = 7000;
+  pgPort = 5432;
+  urlRoot = "http://localhost:${toString port}";
+  user = "pr-tracker";
+in
+  pkgs.nixosTest {
+    name = "api module test";
+
+    nodes.pr_tracker_api = {
+      pkgs,
+      config,
+      ...
+    }: {
+      imports = [modules.api];
+
+      nixpkgs.hostPlatform = pkgs.system;
+
+      services.postgresql.enable = true;
+      services.postgresql.port = pgPort;
+      services.postgresql.ensureDatabases = [user];
+      services.postgresql.ensureUsers = [
+        {
+          name = user;
+          ensureDBOwnership = true;
+        }
+      ];
+
+      services.pr-tracker-api.enable = true;
+      services.pr-tracker-api.port = port;
+      services.pr-tracker-api.user = user;
+      services.pr-tracker-api.databaseUrl = "postgresql:///${user}?host=/run/postgresql&port=${builtins.toString pgPort}";
+      services.pr-tracker-api.localDb = true;
+    };
+
+    testScript = ''
+      pr_tracker_api.start()
+
+      pr_tracker_api.wait_for_unit("pr-tracker-api.service")
+
+      # https://github.com/molybdenumsoftware/pr-tracker/issues/9
+      pr_tracker_api.wait_until_succeeds("curl --fail --connect-timeout 2 ${urlRoot}/api/v1/healthcheck", timeout=60)
```

Comment:
Just do a `wait_for_open_port` in the meantime.

Reply:
Thank you. We did [a thing](https://github.com/molybdenumsoftware/pr-tracker/pull/28), instead.